lib/rust/ensogl/core/src/display/symbol/gpu/registry.rs

```diff
@@ -15,15 +19,43 @@ use crate::display::symbol::WeakSymbol;
 use crate::system::gpu::data::uniform::Uniform;
 use crate::system::gpu::data::uniform::UniformScope;
 use crate::system::gpu::Context;
-
+use crate::system::web;
+use crate::system::web::traits::*;
 
 
 // =============
 // === Types ===
 // =============
 
-pub type SymbolDirty = dirty::SharedSet<SymbolId, Box<dyn Fn()>>;
+pub type Dirty = dirty::SharedSet<SymbolId>;
+
+
+// ===============
+// === RunMode ===
+// ===============
+
+/// The application run mode. It can be set to either [`Normal`] or [`ShaderExtraction`] mode. In
+/// the latter case some warnings are suppressed. For example, there will be no warning that
+/// precompiled shapes shaders were not found, as they are yet to be generated.
+#[derive(Copy, Clone, Debug, Default, Eq, PartialEq)]
+#[allow(missing_docs)]
+pub enum RunMode {
+    #[default]
+    Normal,
+    ShaderExtraction,
+}
```

Comment:
I partially agree with this comment. Ola partially, because `symbol/gpu/registry` is on its way towards becoming top-module "context". It is a global state that we will be referring from more and more places in the app. The refactoring will be done in multiple next PRs and I don't know when it will finish, but basically, the placement of this struct should be in this file, but the file should be in a different place, named differently, and the `Registry` should also be changed. Does it work for you?